ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/StreamsTaskMetadata.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.streams.processor.TaskMetadata;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class StreamsTaskMetadata {
+
+  private final String taskId;
+  private final Set<TopicPartitionEntity> topicPartitions;
```

Comment:
what's the rationale for putting these in separate sets, as opposed to having the `TopicOffset` just have the end committed and end offsets and the TopicPartitionEntity? Then the client doesn't have to think about coherence between the two, and the content is smaller.

Reply:
The is a few cases where the TopicPartition won't have committed yet so I left them separate. We could have a zero default if that would be better

Reply:
I'm in favor of putting them into a single collection with defaults if not committed yet -- could be less vulnerable to bugs that result in inconsistent views.

Reply:
I condensed them into one set that has the TopicPartition, endOffset and the committedOffset